### ISSUE TITLE
[FeatureRequest] Define seperate transformations for train and test data in k-fold.

### ISSUE THREAD

**Issue by mibaumgartner** [delira-dev/delira#35]
The k-fold method in the experiment class currently only works with datamanagers with pre-defined transformations. 
It would be great, if you could define some base-transformations (e.g. pre-processing) which are applied to the train and test set and train-transformations (e.g. augmentation) which are only applied to the training set.

https://github.com/justusschock/delira/blob/3374ac6b86358b6910f3ac1a8c070888e8dc9c33/delira/training/experiment.py#L97-L113
